src/c_api.rs

```diff
@@ -195,46 +200,40 @@ pub extern "C" fn JSONAPI_getStringFromKey(
         let key = unsafe { &mut *key };
         let path = unsafe { CStr::from_ptr(path).to_str().unwrap() };
         if let Ok(s) = key.redis_json.to_string(path, Format::JSON) {
-            if let Ok(s) = CString::new(s) {
-                unsafe {
-                    *str = s.as_bytes_with_nul().as_ptr() as *const c_char;
-                    *len = s.as_bytes().len();
-                }
-                key.cstr_val = Some(s);
-                return Status::Ok as c_int;
-            }
+            set_string(s.as_str(), str, len);
```

Comment:
This, on the other hand, should be serializing (`.to_string(path, Format::JSON`).